### PR TITLE
Provider

### PR DESCRIPTION
**Pull request by a-mabe**
Added
- WorkoutProvider class

Changed
- Utilize WorkoutProvider to load workouts from the database

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -4,14 +4,11 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:google_fonts/google_fonts.dart';
-import 'package:logger/logger.dart';
+// import 'package:logger/logger.dart';
```
Comment (dhruv282) [inline lib/main.dart:7]: should this be removed?

Comment (a-mabe) [reply lib/main.dart:7]: Another PR will be for logging

File: lib/pages/home/home.dart [hunk 2]
```diff
@@ -34,13 +36,16 @@ class _MyHomePageState extends State<MyHomePage> {
   /// The initial list of workouts to be loaded fresh
   /// from the DB.
   ///
-  late Future<List<Workout>> workouts;
+  late Future workouts;
```
Comment (dhruv282) [inline lib/pages/home/home.dart:39]: Shouldn't this stay the same datatype?

File: lib/pages/set_sounds.dart/widgets/sound_dropdown.dart [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:soundpool/soundpool.dart';
-import '../constants/sound_name_map.dart';
+import '../constants/sounds.dart';
```
Comment (dhruv282) [inline lib/pages/set_sounds.dart/widgets/sound_dropdown.dart:3]: wut dis change about?

Comment (a-mabe) [reply lib/pages/set_sounds.dart/widgets/sound_dropdown.dart:3]: leftover junk

File: lib/pages/home/home.dart (hunk 2 above)
Comment (a-mabe) [reply lib/pages/home/home.dart]: Resolved in 7760232ce4de89247f58d4fdcd8cbdcb6b000158

**Review by dhruv282 — APPROVED** [review]
LGTM!

**Pull request merged**